test/chainrules.jl

```diff
@@ -0,0 +1,101 @@
+using ChainRulesCore
+using ChainRulesTestUtils
+
+import MathOptSetDistances
+const MOD = MathOptSetDistances
+const MOI = MathOptSetDistances.MOI
+using FiniteDifferences
+using Test
+using Random
+
+# avoid random finite diff fails because of rounding
+Random.seed!(42)
+
+@testset "rrules multivariate" begin
+    for n in (1, 2, 10)
+        x = randn(n)
+        xb = ChainRulesTestUtils.rand_tangent(x)
+        s = MOI.Reals(n)
+        y = MOD.projection_on_set(MOD.DefaultDistance(), x, s)
+        yb = ChainRulesTestUtils.rand_tangent(y)
+        ChainRulesTestUtils.rrule_test(
+            MOD.projection_on_set,
+            yb,
+            (MOD.DefaultDistance(), nothing),
+            (x, xb),
+            (s, nothing),
+        )
+        s = MOI.Zeros(n)
+        y = MOD.projection_on_set(MOD.DefaultDistance(), x, s)
+        yb = ChainRulesTestUtils.rand_tangent(y)
+        # requires FillArrays.Zero handling
+        # still broken?
+        @test_broken ChainRulesTestUtils.rrule_test(
+            MOD.projection_on_set,
+            yb,
+            (MOD.DefaultDistance(), nothing),
+            (x, xb),
+            (s, nothing),
+        )
+        for s in (MOI.Nonpositives(n), MOI.Nonnegatives(n))
+            y = MOD.projection_on_set(MOD.DefaultDistance(), x, s)
+            yb = ChainRulesTestUtils.rand_tangent(y)
+            ChainRulesTestUtils.rrule_test(
+                MOD.projection_on_set,
+                yb,
+                (MOD.DefaultDistance(), nothing),
+                (x, xb),
+                (s, nothing),
+                atol=10e-5,
```

Comment:
this is the same as 
```suggestion
                atol=1e-4,
```
right?